src/Confluent.Kafka/Consumer.cs

```diff
@@ -1100,19 +1100,35 @@ internal Consumer(ConsumerBuilder<TKey, TValue> builder)
                 };
             }
 
-            TKey key = keyDeserializer != null
-                ? keyDeserializer.Deserialize(rawResult.Key, rawResult.Key == null, new SerializationContext(MessageComponentType.Key, rawResult.Topic))
-                : asyncKeyDeserializer.DeserializeAsync(new ReadOnlyMemory<byte>(rawResult.Key), rawResult.Key == null, new SerializationContext(MessageComponentType.Key, rawResult.Topic))
-                    .ConfigureAwait(continueOnCapturedContext: false)
-                    .GetAwaiter()
-                    .GetResult();
+            TKey key;
+            try
+            {
+                key = keyDeserializer != null
```

Comment:
These layered ternary if statements do not improve readability

Reply:
I find this very easy to read! It's more normal to write things like this in functional languages but C# is pretty functional.